scripts/src/primazactl/primazaworker/workernamespace.py

```diff
@@ -0,0 +1,110 @@
+from primazactl.utils import logger
+from primazactl.kube.namespace import Namespace
+from primazactl.kube.role import Role
+from primazactl.kube.rolebinding import RoleBinding
+from primazactl.kube.roles.primazaroles import get_primaza_namespace_role
+from primazactl.cmd.worker.create.constants import APPLICATION
+from primazactl.primaza.primazacluster import PrimazaCluster
+from primazactl.primazamain.maincluster import MainCluster
+from .workercluster import WorkerCluster
+
+
+class WorkerNamespace(PrimazaCluster):
+
+    main_cluster: str = None
+    type: str = None
+    kube_namesapce: Namespace = None
+    cluster_environment: str = None
+    role_config: str = None
+    main: MainCluster = None
+    worker: WorkerCluster = None
+    secret_cfg: str = None
+
+    def __init__(self, type,
+                 namespace,
+                 cluster_environment,
+                 worker_cluster,
+                 role_config,
+                 main,
+                 worker):
+
+        super().__init__(namespace,
+                         worker_cluster,
+                         f"primaza-{self.type}-agent",
+                         None)
+
+        self.type = type
+        self.main = main
+        self.worker = worker
+        api_client = self.kubeconfig.get_api_client()
+        self.cluster_environment = cluster_environment
+        self.kube_namespace = Namespace(api_client, namespace)
+        self.role_config = role_config
+
+    def create(self):
+        logger.log_entry(f"namespace type: {self.type}, "
+                         f"cluster environment: {self.cluster_environment}, "
+                         f"worker cluster: {self.worker.cluster_name}")
+
+        # On worker cluster
+        # - create the namespace
+        self.kube_namespace.create()
+
+        # Request a new service account from primaza main
+        sa_name = self.main.create_primaza_service_account(
+            self.cluster_environment,
+            self.type)
+
+        # Get kubeconfig with secret from service accounf
+        kc = self.main.get_kubeconfig(sa_name, self.cluster_name)
+
+        # - in the created namespace, create the Secret
+        #     'primaza-auth-$CLUSTER_ENVIRONMENT' the Worker key
+        #     and the kubeconfig for authenticating with the Primaza cluster.
+        if self.type == APPLICATION:
+            secret_name = "primaza-kubeconfig"
+        else:
+            secret_name = "primaza-kubeconfig"
```

Comment:
that seems redundant

Reply:
Yes, moved the name to an object variable so the if..else is not needed at all.